x-pack/plugins/ml/public/application/explorer/explorer.tsx

```diff
@@ -418,6 +427,50 @@ export const Explorer: FC<ExplorerUIProps> = ({
     tableData.anomalies?.length > 0;
 
   const hasActiveFilter = isDefined(swimLaneSeverity);
+  const selectedJobIds = Array.isArray(selectedJobs) ? selectedJobs.map((job) => job.id) : [];
+
+  useEffect(() => {
+    if (!noJobsSelected) {
+      const checkIfSourceIndicesHaveGeoField = async () => {
+        const sourceIndicesWithGeoFieldsMap: {
+          [key: string]: { [key: string]: { geoFields: string[]; dataViewId: string } };
+        } = {};
+        // Go through selected jobs
+        if (Array.isArray(selectedJobs)) {
+          await asyncForEach(selectedJobs, async (job) => {
+            // If job has source indices property
+            if (Array.isArray(job.sourceIndices)) {
+              // Check fields for each index to see if it has geo fields
+              await asyncForEach(job.sourceIndices, async (index) => {
+                const dataViewId = await getDataViewIdFromName(index);
+
+                if (dataViewId) {
+                  const dataView = await dataViewsService.get(dataViewId);
+                  const geoFields = [
+                    ...dataView.fields.getByType(ES_FIELD_TYPES.GEO_POINT),
+                    ...dataView.fields.getByType(ES_FIELD_TYPES.GEO_SHAPE),
+                  ];
+                  if (geoFields.length > 0) {
+                    if (sourceIndicesWithGeoFieldsMap[job.id] === undefined) {
+                      sourceIndicesWithGeoFieldsMap[job.id] = {
+                        [index]: { geoFields: [], dataViewId },
+                      };
+                    }
+                    sourceIndicesWithGeoFieldsMap[job.id][index].geoFields.push(
+                      ...geoFields.map((field) => field.name)
+                    );
+                  }
+                }
+              });
+            }
+          });
+          setSourceIndicesWithGeoFields(sourceIndicesWithGeoFieldsMap);
+        }
+      };
+
+      checkIfSourceIndicesHaveGeoField().catch(console.error); // eslint-disable-line no-console
+    }
+  }, [JSON.stringify(selectedJobIds)]);
```

Comment:
It's worth extracting to a dedicated hook, for instance `useSourceIndicesWithGeoFields`.

Reply:
Is there a benefit to doing that right now since we don't need this functionality anywhere else? 🤔

Reply:
`explorer.tsx` is growing pretty quick :) already 715 LOC

Reply:
Makes sense. Moved out of explorer in 68e56386c7785f7a02c738426a084043212172c6